packages/manager/apps/pci-rancher/src/utils/rancher.ts

```diff
@@ -69,3 +70,80 @@ export const getRancherPlanDescription = (rancherPlan: RancherPlan['name']) => {
       return null;
   }
 };
+
+/**
+ * Extracts drivers and plan information from a switch plan error message.
+ *
+ * @param {string} inputString - The input error message string.
+ * @returns {null|{drivers: string[], plan: string}} - An object containing the drivers and plan, or null if extraction fails.
+ * */
+export function extractDriversAndPlanFromSwitchPlanError(
+  inputString: string,
+): null | { drivers: string[]; plan: string } {
+  try {
+    const openBracketIndex = inputString.indexOf('[');
+    const closeBracketIndex = inputString.indexOf(']');
+    const textBeforeBracket = inputString.substring(0, openBracketIndex).trim();
+    const beginPhrase = 'Unable to switch to plan';
+    if (textBeforeBracket.startsWith(beginPhrase)) {
+      const [plan] =
+        textBeforeBracket.match(/OVHCLOUD_EDITION/) ??
+        textBeforeBracket.match(/STANDARD/);
+      if (!plan) {
+        return null;
+      }
+      const contentInsideBrackets = inputString
+        .substring(openBracketIndex + 1, closeBracketIndex)
+        .trim();
+      if (contentInsideBrackets) {
+        const drivers = contentInsideBrackets.split(',');
+
+        // Create the final array with the text before the bracket and the elements inside the brackets
+        if (drivers.length) {
+          return { drivers, plan };
+        }
+      }
+
+      return null;
+    }
+    return null;
+  } catch (err) {
+    return null;
+  }
+}
+
+/**
+ * Manages Rancher errors and returns appropriate error messages for internationalization.
+ *
+ * @param {Object} error - The error object containing the message and class.
+ * @param {string} error.message - The error message.
+ * @param {string} error.class - The error class.
+ * @returns {[string, TOptions?]} - An array containing the error message key and optional options for internationalization.
+ */
+export const rancherErrorManagement = (error: {
+  message: string;
+  class: string;
+}): [string, TOptions?] => {
+  try {
+    if (error.class === 'Server::InternalServerError') {
+      return ['createRancherErrorInternalServerError'];
+    }
+    if (error.class === 'Client::BadRequest') {
+      const content = extractDriversAndPlanFromSwitchPlanError(error.message);
+      if (content) {
+        const { plan, drivers } = content;
+        return [
+          'createRancherErrorInternalServerBadRequestChangePlan',
+          { plan, drivers: `[${drivers}]` },
+        ];
+      }
+      return [
+        'createRancherError',
+        { rancherCreationErrorMessage: error.message },
+      ];
+    }
+    return [''];
```

Comment:
For readability and better understanding, I think it would be easier to return `null` instead of a single element's tuple containing an empty string as key, what do you think?